### DISCUSSION TITLE
Get-PnpException how to get exception for last command or clear the history

### DISCUSSION THREAD

**Discussion by davepile** [pnp/powershell#1764] [Q&A] [answered]
How can I tell if a command has an exception occur when Get-PnpException provides the last exception regardless of when that happened?

Eg
```
     # command 1
     Connect-PnPOnline -Url $SharePointSiteURL -Credentials $PSCredentials      # assume this encounters an exception
     Get-PnpException              # Shows the exception for Command 1

     # command 2
     Connect-PnPOnline -Url $SharePointSiteURL -Credentials $PSCredentials    # assume this is successful
     Get-PnpException              # Still shows the exception for Command 1
```

The only way I have been able to detect if a Pnp command errors is to capture the output, which makes me wonder when is Get-PnpException of any use.

Is there a way to either clear Get-PnpException history or ideally set each pnp command to clear the history?

**Comment by AndersRask** [comment, marked as answer]
Hey

Usually I use try/catch, but you have to either set
 -Erroraction Stop
For that to work (not all exceptons are terminal) or set error level globally to stop

If you provide a sample of your issue I might better understand your specific issue

Get-PnPException just shows latest error in error stream in a readable format ($Error[0])

**Reply by davepile** [reply]
@AndersRask - Thanks for the reply.

Unfortunately I used a bad example in my post. I only realised after I posted that Connect-PnPOnline does throw an exception.

The initial motivation for my post was that I could not catch an exception caused by Add-PnPFile.

Your answer to set `-Erroraction Stop` is exactly what I was after.

Thank you
